samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoAvailableNotificationServiceImpl.java

```diff
@@ -0,0 +1,369 @@
+            log.debug("Samigo assessment available notification invalidated session.");
+        }
+    }
+
+    private void sendEmailNotification(Site site, PublishedAssessmentFacade publishedAssessment, Member member, ExtendedTime extension) {   // Prep and send the email
+        log.debug("SendEmailNotification: '{}' to {}", publishedAssessment.getTitle(), member.getUserDisplayId());
+        log.debug("Test {}", rl.getString("available_class_at"));
+        String toStr = getUserEmail(member.getUserId());    //begin making the headers
+        String headerToStr = getUserDisplayName(member.getUserId()) + " <" + toStr + ">";
+        String fromStr = "\"" + site.getTitle() + "\" <" + getSetupRequest() + ">";
+        Set<Member> instructors = new HashSet<>();
+        for (Member person : site.getMembers()) {
+            if (StringUtils.equals(person.getRole().getId(), site.getMaintainRole())) {
+                instructors.add(person);
+            }
+        }
+        String replyToStr = getReplyTo(instructors);
+        String subject = rl.getFormattedMessage("email.reminder.subject", site.getTitle(), publishedAssessment.getTitle());
+        StringBuilder body = new StringBuilder();   //start making body text of the email.
+        if(extension != null){  //when this is for an extension, we need to do the date and times differently.
+            body.append(rl.getFormattedMessage("email.reminder.available", publishedAssessment.getTitle(), prettyDate(extension.getStartDate().toInstant())));
+            body.append("<br>");
+            if(extension.getDueDate() != null){
+                body.append(rl.getFormattedMessage("email.reminder.due", prettyDate(extension.getDueDate().toInstant())));
+                body.append("<br>");
+            }
+            if (extension.getTimeHours()==0 && extension.getTimeMinutes()==0) {
+                body.append(rl.getString("email.reminder.nolimit"));
+            } else {
+                body.append(rl.getFormattedMessage("email.reminder.limit", getTimeLimit(convertToSeconds(extension.getTimeHours(),extension.getTimeMinutes()))));
+            }
+        } else {    //normal
+            body.append(rl.getFormattedMessage("email.reminder.available", publishedAssessment.getTitle(), prettyDate(publishedAssessment.getStartDate().toInstant())));
+            body.append("<br>");
+            if(publishedAssessment.getDueDate() != null){
+                body.append(rl.getFormattedMessage("email.reminder.due", prettyDate(publishedAssessment.getDueDate().toInstant())));
+                body.append("<br>");
+            }
+            if (publishedAssessment.getTimeLimit() == 0) {
+                body.append(rl.getString("email.reminder.nolimit"));
+            } else {
+                body.append(rl.getFormattedMessage("email.reminder.limit", getTimeLimit(publishedAssessment.getTimeLimit())));
+            }
+        }
+        //todo: find how to get takebyurl link
```

Comment:
Another TODO here. Are you looking for the published assessment direct URL?

Reply:
I believe we were playing with that idea originally, but decided this worked well enough without it.